src/cluster/src/types.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Types and traits required to setup clusters.
+
+use std::sync::Arc;
+
+use timely::scheduling::SyncActivator;
+use timely::worker::Worker as TimelyWorker;
+
+use mz_cluster_client::client::{ClusterStartupEpoch, TimelyConfig};
+use mz_persist_client::cache::PersistClientCache;
+type ActivatorSender = crossbeam_channel::Sender<SyncActivator>;
+
+pub trait AsRunningWorker<C, R> {
+    fn build_and_run<A: timely::communication::Allocate>(
+        config: Self,
+        timely_worker: &mut TimelyWorker<A>,
+        client_rx: crossbeam_channel::Receiver<(
+            crossbeam_channel::Receiver<C>,
+            tokio::sync::mpsc::UnboundedSender<R>,
+            ActivatorSender,
+        )>,
+        persist_clients: Arc<tokio::sync::Mutex<PersistClientCache>>,
+    );
+}
+
+pub trait CreatableCommand {
+    fn is_create_timely(self) -> Result<(TimelyConfig, ClusterStartupEpoch), Self>
+    where
+        Self: Sized;
+}
+
+impl CreatableCommand for mz_compute_client::protocol::command::ComputeCommand {
+    fn is_create_timely(self) -> Result<(TimelyConfig, ClusterStartupEpoch), Self> {
+        match self {
+            mz_compute_client::protocol::command::ComputeCommand::CreateTimely {
+                config,
+                epoch,
+            } => Ok((config, epoch)),
+            cmd => Err(cmd),
+        }
+    }
+}
```

Comment:
It's not immediately clear what these traits do; they could use docs.

Reply:
Could we also `#[warn(missing_docs)]` on the new crates? It's quite annoying to add that after the fact.

(Since the new crates contain existing code "it's not my job do document other people's code" would be a valid response!)